Strings/Z.cpp

```diff
@@ -0,0 +1,34 @@
+#include<bits/stdc++.h>
+
+using namespace std;
+
+vector<int> get_z(string s) {
+    int n = s.size();
+    vector<int> z(n);
+    int l = 0, r = 0;
+    for (int i = 1; i < n; i++)
+        if (r <= i){
+            l = r = i;
+            while (r < n && s[r] == s[r-l]) r++;
+            z[i] = r-i;
+        }
+        else{
+            if (z[i-l] < (r - i))
+                z[i] = z[i-l];
+            else{
+                l = i;
+                while (r < n && s[r] == s[r-l]) r++;
+                z[i] = r-i;
+            }
+        }
+    return z;
+}
+
+int main(){
```

Comment:
I don't think a main is necessary. When its time to compile the pdf, its much easier to have no extra code in the file. I think we can get rid of the header as well.

Reply:
It's easier to test the code when there's a main. You can also use a main to show how to use the algo.

It's not a big deal for now and if we have a problem with fitting everything in the notebook, we'll deal with it then.